detection_rules/rule.py

```diff
@@ -592,6 +602,24 @@ def get_required_fields(self, index: str) -> List[dict]:
         if validator is not None:
             return validator.get_required_fields(index or [])
 
+    def validate_subclasses(self) -> None:
+        """Check that the query rule subclasses have validate and call method."""
+
+        def validate_recursive(obj: dataclass) -> None:
+            """Recursively navigate through fields and validate if the 'validate' method is present."""
```

Comment:
This is untested, but if we want to avoid recursion we could do something like the following using a stack. That being said, I would almost argue that the recursion is easier to follow. The stack is in effect just getting around the recursive call as logically it is very similar in execution. Could do either way.

```python
def validate_subclasses(self) -> None:
    """Check that the query rule subclasses have validate and call method."""

    # Create a stack with the root object (self)
    stack = [self]

    # Iterate over the stack until it is empty
    while stack:
        # Pop the next object from the stack
        obj = stack.pop()

        # Check if the object is a dataclass
        if is_dataclass(obj):
            # Create a list of field values that need to be validated
            field_values = [getattr(obj, attr.name) for attr in fields(obj) if getattr(obj, attr.name) is not None]
            # Add the field values to the stack for validation
            stack.extend(field_values)

            # Validate the object if it has a 'validate' method
            try:
                obj.validate()
            except AttributeError:
                pass

```